src/com/google/enterprise/gsafeed/GsafeedHelper.java

```diff
@@ -14,177 +14,68 @@
-    UnmarshallerHandler unmarshallerHandler =
-        unmarshaller.getUnmarshallerHandler();
-    xmlReader.setContentHandler(unmarshallerHandler);
-    xmlReader.parse(new InputSource(inputStream));
-    return (Gsafeed) unmarshallerHandler.getResult();
+    return Gsafeed.class.cast(unmarshal(inputStream, Validation.TRUE));
   }
 
   /**
    * Avoid reading the DTD. No validation will happen.
    */
-  public static Gsafeed unmarshalWithoutDtd(URL url) throws JAXBException,
+  public Gsafeed unmarshalWithoutDtd(URL url) throws JAXBException,
       IOException, ParserConfigurationException, SAXException {
     return unmarshalWithoutDtd(url.openStream());
   }
 
   /**
    * Avoid reading the DTD. No validation will happen.
    */
-  public static Gsafeed unmarshalWithoutDtd(InputStream inputStream)
+  public Gsafeed unmarshalWithoutDtd(InputStream inputStream)
       throws JAXBException, IOException, ParserConfigurationException,
       SAXException {
-    SAXParserFactory saxParserFactory = SAXParserFactory.newInstance();
-    saxParserFactory.setValidating(false);
-    saxParserFactory.setXIncludeAware(false);
-    saxParserFactory.setFeature(
-        XMLConstants.FEATURE_SECURE_PROCESSING, true);
-    saxParserFactory.setFeature(
-        "http://xml.org/sax/features/external-general-entities", false);
-    saxParserFactory.setFeature(
-        "http://xml.org/sax/features/external-parameter-entities", false);
-    saxParserFactory.setFeature(
-        "http://apache.org/xml/features/nonvalidating/load-external-dtd",
-        false);
-    XMLReader xmlReader = saxParserFactory.newSAXParser().getXMLReader();
-    xmlReader.setEntityResolver(new EntityResolver() {
-        @Override
-        public InputSource resolveEntity(String publicId,
-            String systemId) throws SAXException, IOException {
-          return new InputSource(new StringReader(""));
-        }
-      });
-    xmlReader.setErrorHandler(new ErrorHandler() {
-        @Override
-        public void warning(SAXParseException exception)
-            throws SAXException {
-          throw exception;
-        }
-
-        @Override
-        public void error(SAXParseException exception)
-            throws SAXException {
-          throw exception;
-        }
-
-        @Override
-        public void fatalError(SAXParseException exception)
-            throws SAXException {
-          throw exception;
-        }
-      });
-    JAXBContext jaxbContext =
-        JAXBContext.newInstance("com.google.enterprise.gsafeed");
-    Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
-    UnmarshallerHandler unmarshallerHandler =
-        unmarshaller.getUnmarshallerHandler();
-    xmlReader.setContentHandler(unmarshallerHandler);
-    xmlReader.parse(new InputSource(inputStream));
-    return (Gsafeed) unmarshallerHandler.getResult();
+    return Gsafeed.class.cast(unmarshal(inputStream, Validation.FALSE));
```

Comment:
An ordinary cast will do now, since we're not trying to cast to a type parameter.